### DISCUSSION TITLE
Custom FavorButton ISSUE

### DISCUSSION THREAD

**Discussion by EnnzoKim** [Favor-Gift-Reminder/Favor-iOS#19] [General]
# ISSUE

![image](https://user-images.githubusercontent.com/97531269/212535911-787009d4-58a0-4889-ba1a-537ff18a6718.png)

- `UIFactory`에 존재하는 `favorButton`은 `disabled`될 때, 위와 같은 버튼의 이미지로 변경됨
- 하지만 `disabled`될 때, 배경색의 이미지는 잘 변하지만, 글씨색도 연하게 되어버리는 이슈 발생 
  - 이 문제는 애플이 버튼 자체를 `disabled`될 때, 일부러 연하게 만든 것 같음
![image](https://user-images.githubusercontent.com/97531269/212536057-9753eb5c-0466-4bad-956c-1325ecf00667.png)

- `Onboarding`에서는 `다음`버튼이 배경색이 연한 버전의 버튼인데 `disabled`일 떄에만 연하게 되어서 버튼 클릭을 받을 수가 없는 문제 발생

# 해결 방안

- `Onboarding UI` 작업이 끝난 내가 이 작업을 수행 
- 버튼을 연한 버전, 진한 버전 버튼을 만들어가지고 `disabled`가 아닌, 두 가지 타입의 Style을 만들어서 버튼을 자체적으로 생성 
- 만약 `disabled`가 필요하다면 특정 객체에서 `isUserInteractionEnabled`을 `False`로 설정하여 따로따로 사용

**Comment by nomatterjun** [comment]
응 나도 `disabled`보다는 다른 스타일의 버튼으로 접근하는게 맞다고 생각해
`FavorButton`에 `highlightedLarge`, `highlightedSmall` 같이 추가해서 작업하면 좋을 것 같아!

`Auth` 쪽은 아직 텍스트필드 작업하느라 조금 남아서 버튼 작업 부탁할게..!

**Reply by EnnzoKim** [reply]
오케이 내가 작업할게 !
